api/src/main.rs

```diff
@@ -546,23 +559,28 @@ async fn main() -> color_eyre::Result<()> {
     tracing::info!("Running SQLx migrations");
     sqlx::migrate!("sqlx/migrations").run(&db).await?;
 
+    let json = ApiDoc::openapi();
     let api = Router::new()
-        .route("/search/list/:name", get(search_list))
-        .route("/search/account/:name", get(search_account))
-        .route("/history/:id", get(history_search))
+        .route("/search/list/{name}", get(search_list))
+        .route("/search/account/{name}", get(search_account))
+        .route("/history/{id}", get(history_search))
         .nest("/list", list::router())
+        .nest("/account", auth::router())
         .nest("/share", share::router())
-        .nest("/account", account::router())
-        .nest("/pantry", pantry::router());
+        .nest("/pantry", pantry::router())
+        .route("/openapi.json", get(|| async { Json(json) }));
 
     #[cfg(feature = "frontend")]
     let frontend = config.frontend.clone();
 
-    let app = Router::new()
-        .merge(SwaggerUi::new("/swagger-ui").url("/api-doc/openapi.json", ApiDoc::openapi()))
+    let app: Router = Router::new()
+        // .merge(Router::<()>::from(SwaggerUi::new("/swagger-ui").url("/api-doc/openapi.json", ApiDoc::openapi()))
         .nest("/api", api)
         .layer(Extension(templates))
         .layer(Extension(db))
+        .layer(Extension(
+            auth::Oauth2Client::from_config(config.clone()).await?,
+        ))
         .layer(
             CorsLayer::new()
                 .allow_origin(config.cors_allow_origin.parse::<HeaderValue>()?)
```

Comment:
It would be better to migrate to using `.state(...)` instead of `Extension`

Reply:
Can't use state as we need to extract the Oauth2Client from unorthodox places (mainly the `auth::User` `FromRequest` implementation )

Reply:
You can do that with State IIRC